UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensor.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using MLAgents.InferenceBrain;
+using UnityEngine;
+
+namespace MLAgents.Sensor
+{
+    public class CameraSensor : SensorBase
+    {
+        public Camera camera;
+        public int width;
+        public int height;
+        public bool grayscale;
+
+        public override int[] GetFloatObservationShape()
+        {
+            return new [] {width, height, grayscale ? 1 : 3};
+        }
+
+        public override void WriteObservation(float[] observationsOut)
+        {
+            throw new NotImplementedException("Have to use compression");
```

Comment:
what about inference?

Reply:
I'm punting on that for now, partly because I'm still trying to decide what the best thing to do about the Y flip is, and how to keep it consistent between training and inference.